cmd/swarm-rafttool/dump.go

```diff
@@ -142,6 +163,33 @@ func dumpSnapshot(swarmdir, unlockKey string) error {
 	}
 	fmt.Println()
 
+	if redact {
+		for _, secret := range s.Store.Secrets {
+			if secret != nil {
+				secret.Spec.Data = []byte("SECRET REDACTED")
+			}
+		}
+		for _, config := range s.Store.Configs {
+			if config != nil {
+				config.Spec.Data = []byte("CONFIG REDACTED")
+			}
+		}
+		for _, task := range s.Store.Tasks {
+			if task != nil {
+				if container := task.Spec.GetContainer(); container != nil {
```

Comment:
Have you confirmed that this actually removes the env vars from the dump? I'm not sure if `task.Spec.GetContainer()` returns a pointer or not.

Reply:
it does return a pointer, and in my WAL dump, the envs is set to "ENVVARS REDACTED" correctly.

Reply:
I'd suggest writing a unit test.

BTW, I recently wrote a redactor that uses struct tags to find out which fields contain sensitive data and need to be redacted. You can give optionally give it a regexp in the struct tag to selectively redact part of a field. I was thinking about open sourcing it; do you think it would be useful here? It would let you annotate the structures with tags (in this case, using the `moretags` gogoprotobuf extension to influence its codegen), and not have to maintain a separate source of truth of which fields are sensitive.